internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/TestBatch.java

```diff
@@ -259,7 +259,7 @@ public void testCompressingRecord(boolean compressionMetricCollectionEnabled, bo
           properties.setProperty(USE_MAPPER_TO_BUILD_DICTIONARY, String.valueOf(useMapperToBuildDict));
         },
         validator,
-        new UpdateStoreQueryParams().setCompressionStrategy(CompressionStrategy.GZIP));
```

Comment:
Do you think it's worth using a dataprovider in this test?  Seems like the code paths would differ significantly enough between these two compression options.

Reply:
I think the intention of this test is for zstd compression, not for gzip since `compressionMetricCollectionEnabled` doesn't apply to gzip compression from my understanding.
@m-nagarajan 
Can you confirm?

Reply:
If `compressionMetricCollectionEnabled == true`, It applies all compression strategies and collects the compressed value sizes using all of them

Reply:
It doesn't matter which strategy is specified for the store. All compression strategies will be applied. So, when this is enabled, the only difference is which compressed value is passed to the reducer.

Reply:
I see, but `useMapperToBuildDict` param doesn't apply if the compression strategy is gzip.

Reply:
Really? Seems like an implementation miss to me, then

Reply:
Isn't it expected? There is no dictionary to build with gzip so why would the extra mapper matter?

Reply:
I think I got the points after the slack conversation.
Once `compressionMetricCollectionEnabled` is enabled, zstd dict build would always happen no matter whether the store setting is using zstd compression strategy or not.
I can revert this change.

Reply:
Or keep the current change since `zstd` compression strategy will test both:
1. Metric collection.
2. Repush with a newly built dict.

What do you think?

Reply:
I am going to resolve this since the change in this test based on the discussion.

Reply:
you can add `testRepush()` in `testZstdCompressingAvroRecordWhenFallbackAvailable` and leave this as `GZIP`?

Reply:
Sounds good.